codeHF/PlotEfficiency_recoStep.C

```diff
@@ -0,0 +1,383 @@
+      TString nameHistgenNonPrompt = outputDir + "/hPtvsYGenNonPrompt"; // generator level pT of generated particles
+      TH2F* hPtvsYGenNonPrompt = (TH2F*)file->Get(nameHistgenNonPrompt.Data());
+      if (!hPtvsYGenNonPrompt) {
+        Printf("Error: Failed to load %s from %s", nameHistgenNonPrompt.Data(), pathFile.Data());
+        return 1;
+      }
+      cgensbys->cd(3);
+      hPtvsYGenNonPrompt->Draw("colz");
+
+      TH1F* hGenPtNonPrompt = (TH1F*)hPtvsYGenNonPrompt->ProjectionX("hGenNonPrompt_Pt", 1, hPtvsYGenNonPrompt->GetXaxis()->GetLast(), "e");
+      TH1F* hGenYNonPrompt = (TH1F*)hPtvsYGenNonPrompt->ProjectionY("hGenNonPrompt_Y", 1, hPtvsYGenNonPrompt->GetYaxis()->GetLast(), "e");
+
+      if (iNRebin > 1) {
+        hGenPtNonPrompt->Rebin(iNRebin);
+        hGenYNonPrompt->Rebin(iNRebin);
+      } else if (NRebin > 1) {
+        hGenPtNonPrompt = (TH1F*)hGenPtNonPrompt->Rebin(NRebin, "hGenPtNonPromptR", dRebin);
+        hGenYNonPrompt = (TH1F*)hGenYNonPrompt->Rebin(NRebin, "hGenYNonPromptR", dRebin);
+      }
+
+      cgensbys_pt->cd(3);
+      hGenPtNonPrompt->Draw("pe");
+
+      cgensbys_y->cd(3);
+      hGenYNonPrompt->Draw("pe");
+
+      for (int iRs = 0; iRs < NRecoStep; iRs++) {
+        //inclusive
+        TString nameHistRec = outputDir + "/hPtvsYRecSig_" + recostep[iRs]; // reconstruction level pT of matched candidates
+        cout << nameHistRec.Data() << endl;
+
+        TH2F* hPtvsYRecIncl = (TH2F*)file->Get(nameHistRec.Data());
+        if (!hPtvsYRecIncl) {
+          Printf("Error: Failed to load %s from %s", nameHistRec.Data(), pathFile.Data());
+          return 1;
+        }
+        hPtvsYRecIncl->SetTitle(Form("k%s", recostep[iRs].Data()));
+        crecosbys->cd(iRs + 1);
+        hPtvsYRecIncl->Draw("colz");
+
+        TH1F* hRecoPtIncl = (TH1F*)hPtvsYRecIncl->ProjectionX(Form("h%s_Pt", recostep[iRs].Data()), 1, hPtvsYRecIncl->GetXaxis()->GetLast(), "e");
+        crecosbys_pt->cd(iRs + 1);
+        hRecoPtIncl->Draw("pe");
+
+        TH1F* hRecoYIncl = (TH1F*)hPtvsYRecIncl->ProjectionY(Form("h%s_Y", recostep[iRs].Data()), 1, hPtvsYRecIncl->GetYaxis()->GetLast(), "e");
+        crecosbys_y->cd(iRs + 1);
+        hRecoYIncl->Draw("pe");
+
+        if (iNRebin > 1) {
+          hRecoPtIncl->Rebin(iNRebin);
+          hRecoYIncl->Rebin(iNRebin);
+        } else if (NRebin > 1) {
+          hRecoPtIncl = (TH1F*)hRecoPtIncl->Rebin(NRebin, "hRecoPtInclR", dRebin);
+          hRecoYIncl = (TH1F*)hRecoYIncl->Rebin(NRebin, "hRecoYInclR", dRebin);
+        }
+
+        auto padEffSbyS_pt = canEffSbyS_pt->cd(1);
+
+        TH1F* hEffPtIncl = nullptr;
+        hEffPtIncl = (TH1F*)hRecoPtIncl->Clone(Form("hEffPtIncl%s", recostep[iRs].Data()));
+        hEffPtIncl->Divide(hEffPtIncl, hGenPtIncl, 1., 1., "B");
+        hEffPtIncl->SetTitle("inclusive ;#it{p}^{rec.}_{T} (GeV/#it{c}); efficiency");
+        yMin = hEffPtIncl->GetMinimum(0) * 0.2;
+        yMax = hEffPtIncl->GetMaximum() * 1.5;
```

Comment:
Why do you multiply the range limits?

Reply:
For visibility. Using only yMin and yMax to fix the range of the histogram axes makes the lowest/highest points closed to the axes. So I decided to use this trick that we already used in HF codes in the past.

Reply:
The margins are already defined with `marginRLow`, `marginRHigh`. They guarantee the size of the vertical margins as a fraction of the pad that height. Why would you need further adjustments?

Reply:
If I do not scale yMin and yMax the plots is too stiched to the axis, it is just a scaling and I didn't think it was a big issue.

Reply:
I don't see why you don't just set the margins the way you want. Scaling `yMin` and `yMax` by constants does not guarantee you any margin size at all as the resulting distance from the axis depends also on the difference/ratio of `yMin` and `Max`.

Reply:
In my case things work fine with the scaling, while I have problems with the margins. Since it is the only remaining issue, I will remove this scaling for the PR and just use it privately.

Reply:
@vkucera I have re-arranged the code in order to fix the issue of `yMin`and `yMax` in the case of step-by-step efficiencies. Now the `yMin` is really the lowest efficiency at all as well as `yMax` the highest at all among all the steps.